test/run-gen-spec-js.py

```diff
@@ -0,0 +1,97 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import os
+import subprocess
+import sys
+
+import find_exe
+import utils
+from utils import Error
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+GEN_SPEC_JS_PY = os.path.join(SCRIPT_DIR, 'gen-spec-js.py')
+
+
+def main(args):
+  parser = argparse.ArgumentParser()
+  parser.add_argument('-o', '--out-dir', metavar='PATH',
+                      help='output directory for files.')
+  parser.add_argument('--wast2wasm', metavar='PATH',
+                      help='override wast2wasm executable.')
+  parser.add_argument('--wasm2wast', metavar='PATH',
+                      help='override wasm2wast executable.')
+  parser.add_argument('--js', metavar='PATH',
+                      help='specify JavaScript compiler to use.')
+  parser.add_argument('--js-flags', metavar='FLAGS',
+                      help='additional flags for JavaScript compiler.',
+                      action='append', default=[])
+  parser.add_argument('--prefix-js',
+                      help='Prefix JavaScript file to pass to gen-spec-js')
+  parser.add_argument('-v', '--verbose', help='print more diagnotic messages.',
+                      action='store_true')
+  parser.add_argument('--no-error-cmdline',
+                      help='don\'t display the subprocess\'s commandline when' +
+                          ' an error occurs', dest='error_cmdline',
+                      action='store_false')
+  parser.add_argument('-p', '--print-cmd', help='print the commands that are run.',
+                      action='store_true')
+  parser.add_argument('--use-libc-allocator', action='store_true')
+  parser.add_argument('file', help='wast file.')
+  options = parser.parse_args(args)
+
+  with utils.TempDirectory(options.out_dir, 'run-gen-spec-js-') as out_dir:
+    wast2wasm = utils.Executable(
+        find_exe.GetWast2WasmExecutable(options.wast2wasm),
+        '--spec',
+        '--no-check-assert-invalid',
+        error_cmdline=options.error_cmdline)
+    wast2wasm.AppendOptionalArgs({
+      '-v': options.verbose,
+      '--use-libc-allocator': options.use_libc_allocator
+    })
+
+    gen_spec_js = utils.Executable(
+        sys.executable, GEN_SPEC_JS_PY,
+        '--temp-dir', out_dir,
+        error_cmdline=options.error_cmdline)
+    gen_spec_js.AppendOptionalArgs({
+      '--wast2wasm': options.wast2wasm,
+      '--wasm2wast': options.wasm2wast,
+      '--prefix': options.prefix_js,
+    })
+    gen_spec_js.verbose = options.print_cmd
+
+    json_file = utils.ChangeDir(utils.ChangeExt(options.file, '.json'), out_dir)
+    js_file = utils.ChangeExt(json_file, '.js')
+    wast2wasm.RunWithArgs(options.file, '-o', json_file)
+
+    if options.js:
+      js = utils.Executable(options.js, *options.js_flags)
+      js.RunWithArgs(js_file)
+      gen_spec_js.RunWithArgs(json_file, '-o', js_file)
```

Comment:
Looks like this execution of these two backwards.. which probably means the --js flag is not used .. is it needed?